### ISSUE TITLE
[HOLD] [$250] Android - Workspace settings - Placeholder "Name" in the field flickers when saving the changes after coming out from offline

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#9938]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to the app and login
2. Create a New Workspace if it doesn't exist yet
3. Go to Settings / Workspace settings
4. Disable the internet connection on the device
5. Change the workspace currency and name
6. Save the changes
7. Go back online

## Expected Result:
Placeholder "name" in the name field doesn't flicker or move

## Actual Result:
Placeholder "name" in the name field flickers and can see a jerk

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Android

**Version Number:** v1.1.84-11

**Reproducible in staging?:**  Y

**Reproducible in production?:**  Y

**Email or phone of affected tester (no customers):** 
applausetester+1507@applause.expensifail.com / Yulia1986Gerets

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/179226246-d6b1268e-6aff-445c-b5c1-2da220817313.mp4

<img src="https://user-images.githubusercontent.com/43996225/179226465-a0388372-3249-4433-a22f-1fb6732e980c.jpg" width="40%" height="40%">

**Expensify/Expensify Issue URL:** 

**Issue reported by:** 
Applause internal team
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by bfitzexpensify** [comment]
Upwork job [here](https://www.upwork.com/jobs/~013ca8b17a0e2d48a4)

**Comment by MananGadhiyaSureshbhai** [comment]
@kavimurut he issue is reproducing across the app not only for general settings

**Comment by errahsoufiane** [comment]
The issue happens from this [component](https://github.com/Expensify/App/blob/main/src/components/TextInput/TextInputLabel/index.native.js)

label width is 0 (initial state value) before onLayout callback and it flickers when render with active label scale.
2nd parameter of textInputLabelTransformation is 0 before width is set from onLayout callback.

In summary, flicker happens when label view scaled to 0.8668(ACTIVE_LABEL_SCALE) and width = 0
Label goes back to correct place when width set correctly(nativeEvent.layout.width) after onLayout callback

My suggestion to fix: hide label until width state value is set (update this [code](https://github.com/Expensify/App/blob/main/src/components/TextInput/TextInputLabel/index.native.js#L33))
before: {this.props.label}
after: {this.state.width > 0 && this.props.label}
I tested and confirmed this fix on both platforms (iOS, android)

**Comment by tgolen** [comment]
I like that solution! 👍 🟢

**Comment by parasharrajat** [comment]
@errahsoufiane Do you have any other suggestions apart from the other? Although it is doable but does not give the best UI.

**Comment by errahsoufiane** [comment]
I already tried to show label even if width is not set yet. But based on current transform implementation of this component, it's impossible to set x coordinate correctly without knowing the width when view scaled to ACTIVE_LABEL_SCALE.
But this is very rare case and in very short time, when app becomes a little heavy due to some background tasks. Mostly not happen so I think it doesn't affect user experience much.

**Comment by parasharrajat** [comment]
I did some testing and you might be correct as it will be hardly visible. When the lag is visible, onLayout takes about 300ms to fire after mounting (On Dev). Looking at a different perspective, do you think we can get away from this transform implementation so we don't have to deal with this issue at all. 

The next thing would be to get some videos in and test the whole app to see if the hiding labels do not have a bad effect on any page. This video will be used to verify the UI by the designer team.

**Comment by errahsoufiane** [comment]
https://user-images.githubusercontent.com/108141252/180061013-a0159aa6-3a7c-4f54-bd79-daf9712420f4.mp4

https://user-images.githubusercontent.com/108141252/180061700-785eb616-db33-40eb-95dd-23863e73f5bf.mp4

**Comment by bfitzexpensify** [comment]
@parasharrajat it's a little unclear above - are you 👍 for this proposal, or are we still ironing out the details?

**Comment by bfitzexpensify** [comment]
Bump @parasharrajat

**Comment by shawnborton** [comment]
Should we pause on this one while we finish up offline improvements?

**Comment by errahsoufiane** [comment]
> Should we pause on this one while we finish up offline improvements?

This happens not only when offline/online switch but also when app becomes heavy due to many background tasks (I confirmed)

**Comment by tgolen** [comment]
Hm, it's possible that the offline refactorings could have an effect on this (because it _might_ decrease the number of background tasks). It feels like this issue is such an edge case and low value, that I don't mind putting a HOLD on it for a couple of weeks to re-evaluate it later.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by bfitzexpensify** [comment]
OK, given @tgolen's comment:

>this issue is such an edge case and low value, that I don't mind putting a HOLD on it for a couple of weeks to re-evaluate it later.

And the fact it was not reproducible during the KI test, I agree with putting this on HOLD and seeing whether it still occurs after offline first is finished in a few weeks. Doing that now - let me know if anyone disagrees with that approach

**Comment by tgolen** [comment]
Gonna drop this down to monthly for the time being then.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week) 
@tgolen ok to Close?

**Comment by tgolen** [comment]
Yes, thank you for testing it again! I'll go ahead and close it out.
